roles/backup/tasks/postgres.yml

```diff
@@ -82,6 +82,15 @@
     resolvable_db_host: '{{ (awx_postgres_type == "managed") | ternary(awx_postgres_host + "." + ansible_operator_meta.namespace + ".svc.cluster.local", awx_postgres_host) }}'  # yamllint disable-line rule:line-length
   no_log: "{{ no_log }}"
 
+- name: Precreate database partitions
+  k8s_exec:
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    pod: "{{ awx_task_pod_name }}"
+    container: "{{ ansible_operator_meta.name }}-task"
+    command: awx-manage precreate_partitions --count='{{ precreate_partition_hours }}'
+  register: result
+  changed_when: "'Created partitions for' in result.stdout"
+
```

Comment:
do not run this block when precreate_partition_hours==0

Reply:
example https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_conditionals.html#basic-conditionals-with-when